inference-engine/src/mkldnn_plugin/nodes/mkldnn_color_convert_node.cpp

```diff
@@ -0,0 +1,666 @@
+}
+
+const void * MKLDNNColorConvertNode::Converter::input(size_t idx) const {
+    return _node->getParentEdgeAt(idx)->getMemoryPtr()->GetPtr();
+}
+
+void * MKLDNNColorConvertNode::Converter::output(size_t idx) const {
+    return _node->getChildEdgeAt(idx)->getMemoryPtr()->GetPtr();
+}
+
+const VectorDims & MKLDNNColorConvertNode::Converter::inputDims(size_t idx) const {
+    return _node->getParentEdgesAtPort(idx)[0]->getMemory().getStaticDims();
+}
+
+MKLDNNColorConvertNode::MKLDNNColorConvertNode(const std::shared_ptr<ngraph::Node>& op,
+                                               const mkldnn::engine& eng,
+                                               MKLDNNWeightsSharing::Ptr &cache)
+    : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    std::tie(algorithm, errorMessage) = getAlgorithmFor(op);
+    if (algorithm == Algorithm::Default)
+        IE_THROW(NotImplemented) << errorMessage;
+}
+
+void MKLDNNColorConvertNode::getSupportedDescriptors() {}
+
+void MKLDNNColorConvertNode::initSupportedPrimitiveDescriptors() {
+    if (supportedPrimitiveDescriptors.empty()) {
+        switch (algorithm) {
+            case Algorithm::NV12toRGB:
+            case Algorithm::NV12toBGR: {
+                for (const auto &desc : nv12::supportedPrimitives(this))
+                    addSupportedPrimDesc(std::get<0>(desc), std::get<1>(desc), std::get<2>(desc), std::get<3>(desc));
+                initSupportedNV12Impls();
+                break;
+            default:
+                break;
+            }
+        }
+    }
+}
+
+void MKLDNNColorConvertNode::initSupportedNV12Impls() {
+    #define SUPPORTED_IMPL(Impl, type, desc_type)                           \
+        [](MKLDNNNode *node) {                                              \
+            return new nv12::Impl<type, impl_desc_type::desc_type>(node);   \
+        };
+
+    // ref
+    {
+        auto &impls = _supportedImpls[impl_desc_type::ref][algorithm];
+        impls[Precision::U8][true] = SUPPORTED_IMPL(SinglePlaneConvert, uint8_t, ref);
+        impls[Precision::U8][false] = SUPPORTED_IMPL(TwoPlaneConvert, uint8_t, ref);
+        impls[Precision::FP32][true] = SUPPORTED_IMPL(SinglePlaneConvert, float, ref);
+        impls[Precision::FP32][false] = SUPPORTED_IMPL(TwoPlaneConvert, float, ref);
+    }
+
+    // jit_uni
+    {
+        auto &impls = _supportedImpls[impl_desc_type::jit_uni][algorithm];
+        impls[Precision::U8][true] = SUPPORTED_IMPL(SinglePlaneConvert, uint8_t, jit_uni);
+        impls[Precision::U8][false] = SUPPORTED_IMPL(TwoPlaneConvert, uint8_t, jit_uni);
+        impls[Precision::FP32][true] = SUPPORTED_IMPL(SinglePlaneConvert, float, jit_uni);
+        impls[Precision::FP32][false] = SUPPORTED_IMPL(TwoPlaneConvert, float, jit_uni);
+    }
+
+    #undef SUPPORTED_IMPL
+}
+
+void MKLDNNColorConvertNode::createPrimitive() {
+    prepareParams();
+}
+
+void MKLDNNColorConvertNode::execute(mkldnn::stream strm) {
+    _impl->execute(strm);
```

Comment:
I guess with this implementation we have to always check, that _impl is initialized, and throw if not.
Otherwise I can imagine that for example, clockwork, would complain.

Reply:
Added check for null pointer